File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
       for(int i=0,j=s.length()-1;i<j;){
           if(!Character.isLetterOrDigit(s.charAt(i))) i++;
           else if(!Character.isLetterOrDigit(s.charAt(j)))j--;
           else if(Character.toLowerCase(s.charAt(i++))!=Character.toLowerCase(s.charAt(j--))) return false;
       }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str = "";
        for(char c:s){
            if(iswalnum(c)){
                str += tolower(c);
            }
        }
        int i=0, j=str.length()-1;
        while(i<j){
            if(str[i]!=str[j]){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};